x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -724,4 +706,10 @@ private static boolean verifyBcryptHash(SecureString text, char[] hash) {
         SECURE_RANDOM.nextBytes(salt);
         return salt;
     }
+
+    private static char[] hashSha512NoSalt(SecureString text) {
```

Comment:
nit: 
```suggestion
    private static char[] sha512Hash(SecureString text) {
```

(hashing doesn't inherently include a salt, so it feels strange calling it out in the method name )